extensions/amp-form/0.1/amp-form.js

```diff
@@ -567,12 +571,21 @@ export class AmpForm {
       container.setAttribute('role', 'alert');
       container.setAttribute('aria-labeledby', messageId);
       container.setAttribute('aria-live', 'assertive');
-      return this.templates_.findAndRenderTemplate(container, data)
-          .then(rendered => {
-            rendered.id = messageId;
-            rendered.setAttribute('i-amphtml-rendered', '');
-            container.appendChild(rendered);
-          });
+
+      if (this.templates_.hasTemplate(container)) {
+        this.templates_.findAndRenderTemplate(container, data)
+            .then(rendered => {
+              rendered.id = messageId;
+              rendered.setAttribute('i-amphtml-rendered', '');
+              container.appendChild(rendered);
+            });
+      } else {
+        // TODO(vializ): This will be unnecessary when the AMP Layers
+        // implementation is complete. We call mutateElement here and not
+        // where the template is made visible so that we don't do
+        // redundant layout work when a template is rendered too.
+        this.resources_.mutateElement(container, () => {});
```

Comment:
let's add a test for it in `amp-integration-form.js`. It would be similar to `should submit and render error` test, except no `template` and let's do text and an `amp-img`.  To verify that the `amp-img` got layedout correctly, you can assert that there exists an `img` element inside `amp-img` after message is rendered.